submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionPersistenceService.kt

```diff
@@ -0,0 +1,43 @@
+package ac.uk.ebi.biostd.persistence.service
+
+import ac.uk.ebi.biostd.persistence.integration.FileMode
+import ac.uk.ebi.biostd.persistence.mapping.extended.to.ToExtSubmissionMapper
+import ac.uk.ebi.biostd.persistence.model.DbSubmission
+import ac.uk.ebi.biostd.persistence.repositories.SubmissionDataRepository
+import ac.uk.ebi.biostd.persistence.repositories.UserDataDataRepository
+import ac.uk.ebi.biostd.persistence.repositories.data.SubmissionRepository
+import ac.uk.ebi.biostd.persistence.service.filesystem.FileSystemService
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.model.constants.ProcessingStatus.PROCESSED
+import ebi.ac.uk.model.constants.ProcessingStatus.PROCESSING
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+
+open class SubmissionPersistenceService(
+    private val subRepository: SubmissionRepository,
+    private val subDataRepository: SubmissionDataRepository,
+    private val userDataRepository: UserDataDataRepository,
+    private val systemService: FileSystemService,
+    private val toExtSubmissionMapper: ToExtSubmissionMapper
+) {
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW, readOnly = false)
+    open fun processSubmission(submission: ExtSubmission, mode: FileMode): ExtSubmission {
+        systemService.persistSubmissionFiles(submission, mode)
```

Comment:
I think it would be nice to add break lines to this method separating the actions performed. It looks a bit chunky.